Log Watchdog events to scuba

bchen2020 opened the pull request:
Summary: This diff logs some events of FileTimerServer to a scuba table. The events include "server started", "server stopped", "set timer", "clear timer" and "kill worker process".

Test Plan:
### Unit Test
```
buck test mode/dev-nosan //caffe2/test/distributed/elastic/agent/server/test:local_agent_test
```
```
Test Session: https://www.internalfb.com/intern/testinfra/testrun/1407375146936031
RE: reSessionID-2224cf79-6a28-4762-ab7c-9875adb244dc 3.4 KiB▲,  0.0 B▼
Jobs completed: 57. Time elapsed: 3084.4s.
Tests finished: Pass 55. Fail 0. Fatal 0. Skip 0. 0 builds failed
```

Differential Revision: D39665560

H-Huang commented on test/distributed/elastic/agent/server/test/local_elastic_agent_test.py:
```diff
@@ -32,8 +33,9 @@
-    TORCHELASTIC_TIMER_FILE,
+    TORCHELASTIC_ENABLE_FILE_TIMER,
 )
+from torch.distributed.elastic.events.fb.scuba import ScubaLogHandler
```
I'm surprised we don't see any failures from this. Are torch.elastic tests even running in our OSS CI? How can we be importing from internal fb-only code in our open source mirror? cc @d4l3k

d4l3k replied:
@H-Huang I'm not 100% sure -- how are are the builds configured in OSS CI? / how do I check? 

@kiukchung would probably have the best context here

bchen2020 replied:
I moved the scuba test to a FB internal test module.

kiukchung replied:
the only tests we "blocklist" are defined [here](https://github.com/pytorch/pytorch/blob/master/test/run_test.py#L95), which doesn't list `distributed.elastic.agent.server.test.local_elastic_agent_test`. Did the CI that was triggered for this PR not run `local_elastic_agent_test.py`?

H-Huang reviewed (changes requested):
Curious about how scuba logging works with open source changes. Left a comment.

bchen2020 committed:
Log Watchdog events to scuba (#85391)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/85391

This diff logs some events of FileTimerServer to a scuba table. The events include "server started", "server stopped", "set timer", "clear timer" and "kill worker process".

Test Plan:
### Unit Test
```
buck test mode/dev-nosan //caffe2/test/distributed/elastic/agent/server/test:local_agent_test
```
```
Test Session: https://www.internalfb.com/intern/testinfra/testrun/3377699847897862
RE: reSessionID-3c80bc58-9ae9-415e-9c08-eb2673a2741e 45.4 MiB▲,  2.0 GiB▼
Jobs completed: 180453. Time elapsed: 3242.8s. Cache hits: 99%. Commands: 71541 (cached: 71533, remote: 4, local: 4)
Tests finished: Pass 55. Fail 0. Fatal 0. Skip 0. 0 builds failed
```
--------
```
buck test mode/dev-nosan //caffe2/test/distributed/elastic/agent/server/test/fb:local_agent_fb_internal_test
```
```
Test Session: https://www.internalfb.com/intern/testinfra/testrun/6192449561563806
RE: reSessionID-e71503c2-606e-4b10-ac16-3772184fae5e 246.9 KiB▲,  1.1 MiB▼
Jobs completed: 47. Time elapsed: 168.6s. Cache hits: 0%. Commands: 11 (cached: 0, remote: 11, local: 0)
Tests finished: Pass 2. Fail 0. Fatal 0. Skip 0. 0 builds failed
```

This query (https://fburl.com/scuba/pytorch_elastic_status/tvugvh17) shows some events are written to the scuba table `pytorch_elastic_status` .

Reviewed By: d4l3k

Differential Revision: D39665560

fbshipit-source-id: f213abbaebacef38bb0a03060666e7f1c2bb5492

bchen2020 commented:
> 



> Curious about how scuba logging works with open source changes. Left a comment.

The _log_handlers in open source maps the "null" destination to a logging.NullHandler, while with FB internal code, it maps to a scuba log handler.

d4l3k reviewed (approved):
would be nice to have logging test in OSS
